### DISCUSSION TITLE
Loading localized messages for `react-intl` in a `pages` project

### DISCUSSION THREAD

**Discussion by imageck** [vercel/next.js#50061] [Help]
### Summary

Hi.

I'm using `react-intl` in a `pages` project. After a deep and scrupulous research, I settled on dynamically `import`ing inside `getInitialProps()`:
```jsx
MyApp.getInitialProps = async (ctx) => {
  let { locale, defaultLocale } = ctx.ctx;
  let props = await App.getInitialProps(ctx);
  let messages = await import(`messages/compiled/${locale ?? defaultLocale}.json`);
  return { ...props, messages };
};
```

That obviously disables ASO, which requires exporting a `getStaticProps()` function in any page that needs static generation.

I'd really like to find a better alternative to `getInitialProps()` before the project grows in size. I've seen some load translations in either `useEffect()` or `useMemo()` but I'm not sure how that would work. Also, it is a plain JS project, so the `AppProps` type and its built-in `appInitialProps` goes out the window.

### Additional information

```js
next@13.3.1
react-intl@6.4.1
```

**Comment by sergioviniciuss** [comment]
it might be a repetitive approach, but you could actually wrap pages with intlProvider and avoid losing ASO.

I created page templates, so similar pages can have the same "wrapper" that has the provider so... 

I can't confirm this is the best approach, but from my experience it's what I could recommend.
